internal/services/securitycenter/security_center_subscription_pricing_resource.go

```diff
@@ -134,8 +134,10 @@ func resourceSecurityCenterSubscriptionPricingUpdate(d *pluginsdk.ResourceData,
 	}
 
 	extensionsStatusFromBackend := make([]pricings_v2023_01_01.Extension, 0)
-	if err == nil && apiResponse.Model != nil && apiResponse.Model.Properties != nil && apiResponse.Model.Properties.Extensions != nil {
-		extensionsStatusFromBackend = *apiResponse.Model.Properties.Extensions
+	if err == nil && apiResponse.Model != nil && apiResponse.Model.Properties != nil {
+		if apiResponse.Model.Properties.Extensions != nil {
+			extensionsStatusFromBackend = *apiResponse.Model.Properties.Extensions
+		}
```

Comment:
Given that `extensionsStatusFromBackend` will always be nil on the first apply here, we should call `client.Update` twice. On the first `client.Update` we create the resource as usual and use the response from this to populate `extensionsStatusFromBackend`. On the second `client.Update` we can now update the extensions.